src/VaultSharp/V1/SecretsEngines/Identity/IIdentitySecretsEngine.cs

```diff
@@ -8,6 +8,74 @@ namespace VaultSharp.V1.SecretsEngines.Identity
     /// </summary>
     public interface IIdentitySecretsEngine
     {
+        /// <summary>
+        /// Create or update a role. ID tokens are generated against a role and
+        /// signed against a named key.
+        /// </summary>
+        /// <param name="roleName">Name of the role.</param>
+        /// <param name="createRoleRequest">Request object ot create or update a role.</param>
+        /// <param name="mountPoint"><para>[optional]</para>
+        /// The mount point for the backend. Defaults to <see cref="SecretsEngineMountPoints.Identity" />
+        /// Provide a value only if you have customized the Azure mount point.
+        /// </param>
+        /// <returns>This endpoint doesn't return anything.</returns>
+        Task CreateRoleAsync(string roleName, CreateRoleRequest createRoleRequest, string mountPoint = null);
+
+        /// <summary>
+        /// This endpoint queries a role and returs its configuration.
+        /// </summary>
+        /// <param name="roleName">Name of the role.</param>
+        /// <param name="mountPoint"><para>[optional]</para>
+        /// The mount point for the backend. Defaults to <see cref="SecretsEngineMountPoints.Identity" />
+        /// Provide a value only if you have customized the Azure mount point.
+        /// </param>
+        /// <returns>The role information</returns>
+        Task<ReadRoleResponse> ReadRoleAsync(string roleName, string mountPoint = null);
+
+        /// <summary>
+        /// This endpoint deletes a role.
+        /// </summary>
+        /// <param name="roleName">Name of the role.</param>
+        /// <param name="mountPoint"><para>[optional]</para>
+        /// The mount point for the backend. Defaults to <see cref="SecretsEngineMountPoints.Identity" />
+        /// Provide a value only if you have customized the Azure mount point.
+        /// </param>
+        /// <returns>This endpoint doesn't return anything</returns>
+        Task DeleteRoleAsync(string roleName, string mountPoint = null);
+
+        /// <summary>
+        /// This endpoint creates or updates a named key which is used by a role to sign tokens.
+        /// </summary>
+        /// <param name="name">Name of the named key.</param>
+        /// <param name="mountPoint"><para>[optional]</para>
+        /// The mount point for the backend. Defaults to <see cref="SecretsEngineMountPoints.Identity" />
+        /// Provide a value only if you have customized the Azure mount point.
+        /// </param>
+        /// <param name="createNamedKeyRequest">Request object to create a named key.</param>
+        /// <returns>This endpoint doesn't return anything.</returns>
+        Task CreateNamedKeyAsync(string name, CreateNamedKeyRequest createNamedKeyRequest, string mountPoint = null);
```

Comment:
string keyName

Reply:
Hey @rajanadar, 
I have updated the `name` parameter to `keyName` in all the interface methods and implementations w.r.t the Named keys in the OIDC Identity Provider.